### PR TITLE
fix: merge .percy.yml config options with snapshot options for serializeDOM

### PR DESCRIPTION
**Pull request by rishigupta1599**
## Summary
- Config options from `.percy.yml` were not being passed to `PercyDOM.serialize()`
- Only per-snapshot options were used for DOM serialization
- Adds `MergeSnapshotOptions()` helper that extracts `cliConfig.snapshot` properties and merges with user options, with per-snapshot options taking priority

## Test plan
- [ ] Existing tests pass
- [ ] Verify `.percy.yml` config options are applied during DOM serialization

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Percy/Percy.cs [hunk 1]
```diff
@@ -728,6 +728,34 @@ internal static List<object> CaptureResponsiveDom(IPage page, Dictionary<string,
...
+                snapshotElement.ValueKind == JsonValueKind.Object)
+            {
+                foreach (JsonProperty prop in snapshotElement.EnumerateObject())
+                {
+                    merged[prop.Name] = prop.Value.ValueKind switch
+                    {
+                        JsonValueKind.True => true,
+                        JsonValueKind.False => false,
+                        JsonValueKind.Number => prop.Value.TryGetInt32(out int intVal) ? intVal : (object)prop.Value.GetDouble(),
+                        JsonValueKind.String => prop.Value.GetString(),
```
Comment (rishigupta1599) [inline Percy/Percy.cs]: **[Low] Inconsistent null / nested-value handling in MergeSnapshotOptions**

`JsonValueKind.String => prop.Value.GetString()` boxes a nullable string into a `Dictionary<string,object>`, and a JSON `null` value falls to `_ => prop.Value`, storing a raw `JsonElement` of kind `Null`. Either can surprise downstream code that does runtime type dispatch (`is bool` / `is int`). It serializes harmlessly today, but the typing is inconsistent.

**Suggestion:** Skip JSON `null` values (or normalize to a real `null`/empty string) so the merged dictionary holds only type-consistent entries.

_Reviewer: stack:code-review_

File: Percy/Percy.cs [hunk 2]
```diff
@@ -786,12 +814,16 @@ responsiveOptionNoConfig is bool responsiveFromOptionsNoConfig &&
...
+                // Merge .percy.yml config options with snapshot options (snapshot options take priority)
+                var mergedOptions = MergeSnapshotOptions(optionsDict);
+
                 // CookiesAsync has no sync equivalent; block with .GetAwaiter().GetResult() to fetch cookies before serializing the DOM
                 var cookies = page.Context.CookiesAsync().GetAwaiter().GetResult();
                 string cookiesJson = JsonSerializer.Serialize(cookies);
-                object domSnapshot = IsResponsiveSnapshotCapture(optionsDict)
-                    ? CaptureResponsiveDom(page, optionsDict, cookiesJson)
-                    : GetSerializedDom(page, optionsDict, cookiesJson);
+                object domSnapshot = IsResponsiveSnapshotCapture(mergedOptions)
```
Comment (rishigupta1599) [inline Percy/Percy.cs:890]: **[Low] Redundant merged options passed to IsResponsiveSnapshotCapture**

`IsResponsiveSnapshotCapture` already independently reads `cliConfig.snapshot.responsiveSnapshotCapture`. Passing `mergedOptions` (which now also contains that key) makes the config value match via the per-call `options.TryGetValue` branch, leaving the method's own config branch effectively dead when the merge fires. Correct today, but a maintenance trap.

**Suggestion:** Keep passing `optionsDict` here (it handles config internally) and pass `mergedOptions` only to the DOM-serialization calls.

_Reviewer: stack:code-review_

**Review by rishigupta1599 — COMMENTED** [review]
Claude Code Review (automated) — 2 inline finding(s). Full report in the PR comment below. Verdict: Passed.

**Comment by rishigupta1599** [comment]
# Claude Code PR Review

**PR:** https://github.com/percy/percy-playwright-dotnet/pull/220  •  **Head:** 4c1c8cb  •  **Reviewers:** stack:code-review

## Summary
Adds a private `MergeSnapshotOptions()` helper that builds a `Dictionary<string,object>` from `cliConfig.snapshot.*` (coercing JSON primitives to native types) and overlays per-call options on top (per-call priority), then routes the merged result into `IsResponsiveSnapshotCapture`, `CaptureResponsiveDom`, and `GetSerializedDom` so that global `.percy.yml` snapshot config influences DOM serialization (PER-8053).

## Review Table

| Priority | Category | Check | Status | Notes |
|----------|----------|-------|--------|-------|
| High | Security | No hardcoded secrets or credentials | N/A | Security lens skipped per orchestrator policy. |
| High | Security | Authentication/authorization checks present | N/A | Security lens skipped per orchestrator policy. |
| High | Security | Input validation and sanitization | N/A | Security lens skipped per orchestrator policy. |
| High | Security | No IDOR — resource ownership validated | N/A | Security lens skipped per orchestrator policy. |
| High | Security | No SQL injection (parameterized queries) | N/A | Security lens skipped per orchestrator policy. |
| High | Correctness | Logic is correct, handles edge cases | Pass | Merge order (config first, per-call overlay) correctly implements per-call priority. Primitive coercion is sound; nested arrays/objects fall through to raw `JsonElement` which round-trips correctly via `JsonSerializer.Serialize` in `GetSerializedDom`. |
| High | Correctness | Error handling is explicit, no swallowed exceptions | Pass | Method is pure/total; the enclosing `Snapshot()` retains its try/catch. No new swallowed exceptions. |
| High | Correctness | No race conditions or concurrency issues | Pass | Operates on local dictionaries; `cliConfig` is read-only here. No new shared mutable state. |
| Medium | Testing | New code has corresponding tests | Fail | No test in `Percy.Test/` exercises `MergeSnapshotOptions` — neither config-only application, per-call override precedence, nor the null-config path. See Findings. |
| Medium | Testing | Error paths and edge cases tested | Fail | No coverage for null `options`, absent/non-object `cliConfig.snapshot`, or key-collision cases. |
| Medium | Testing | Existing tests still pass (no regressions) | Pass | Change is additive; existing call sites preserved. No evidence of regression. |
| Medium | Performance | No N+1 queries or unbounded data fetching | Pass | Single bounded pass over config props + per-call options. Negligible cost. |
| Medium | Performance | Long-running tasks use background jobs | N/A | Not applicable to this SDK path. |
| Medium | Quality | Follows existing codebase patterns | Pass | Mirrors the existing `JsonElement`/`ValueKind` inspection style used by `IsResponsiveSnapshotCapture`. |
| Medium | Quality | Changes are focused (single concern) | Pass | Single concern: merging config snapshot options before DOM serialization. Touches one file. |
| Low | Quality | Meaningful names, no dead code | Pass | `MergeSnapshotOptions`/`mergedOptions` are clear. Minor redundancy noted in Findings (config re-read inside `IsResponsiveSnapshotCapture`). |
| Low | Quality | Comments explain why, not what | Pass | The "snapshot options take priority" comment documents intent. |
| Low | Quality | No unnecessary dependencies added | Pass | No new dependencies; uses existing `System.Text.Json`. |

## Findings

- **File:** `Percy/Percy.cs:739`
- **Severity:** Medium
- **Reviewer:** stack:code-review
- **Issue:** No tests cover the new `MergeSnapshotOptions` logic. The PR's core invariant (per-call options override config; config-only options flow through to `PercyDOM.serialize`) is untested, as is the null-`cliConfig`/null-`options` path. `Percy.Test/` has integration patterns (config injection + request inspection) that make this straightforward.
- **Suggestion:** Add tests asserting: (a) a config-only `snapshot.enableJavaScript: true` reaches the serialize call when no per-call options are passed; (b) a per-call value overrides the same config key; (c) null/absent `cliConfig.snapshot` yields just the per-call options.

- **File:** `Percy/Percy.cs:746`
- **Severity:** Low
- **Reviewer:** stack:code-review
- **Issue:** `JsonValueKind.String => prop.Value.GetString()` boxes a nullable string into a `Dictionary<string,object>`; a JSON `null` value falls to `_ => prop.Value` and is stored as a raw `JsonElement` of kind `Null`. Either may surprise downstream code that does runtime type dispatch (`is bool` / `is int`). It serializes harmlessly today, but the type handling is inconsistent.
- **Suggestion:** Skip JSON `null` values (or normalize to a real `null`/empty string) so the merged dictionary contains only meaningful, type-consistent entries.

- **File:** `Percy/Percy.cs:824`
- **Severity:** Low
- **Reviewer:** stack:code-review
- **Issue:** Passing `mergedOptions` into `IsResponsiveSnapshotCapture` is redundant: that method already independently reads `cliConfig.snapshot.responsiveSnapshotCapture` (lines 783-790). The merged dict now also contains that key, so the config value is matched via the per-call `options.TryGetValue` branch instead, making the method's own config branch effectively dead when the merge fires. Result is correct today, but the duplicated responsibility is a maintenance trap.
- **Suggestion:** Keep passing `optionsDict` to `IsResponsiveSnapshotCapture` (it handles config internally) and pass `mergedOptions` only to the DOM-serialization calls, preserving separation of concerns.

- **File:** `Percy/Percy.cs:836`
- **Severity:** Low
- **Reviewer:** stack:code-review
- **Issue:** Observation (not a regression from this PR): the request payload sent to `/percy/snapshot` is still built from the original per-call `options`, not `mergedOptions`, so config snapshot options affect local DOM serialization but are not forwarded in the request. This is consistent with PER-8053's "before serializeDOM" scope (the CLI merges its own `.percy.yml` server-side), so it appears intentional. Separately, `snapshotOptions.Add()` would throw `ArgumentException` if a user config key ever collided with a reserved key (`name`, `url`, etc.) — a pre-existing edge case, unchanged by this PR.
- **Suggestion:** Confirm the request-payload behavior is intended; if config keys should also be forwarded, switch the loop to `mergedOptions` using indexer assignment (`snapshotOptions[k] = v`) rather than `.Add()` to avoid collisions.

---

**Verdict: PASS**

File: Percy/Percy.cs [hunk 3]
```diff
@@ -728,6 +728,34 @@ internal static List<object> CaptureResponsiveDom(IPage page, Dictionary<string,
...
+                configElement.TryGetProperty("snapshot", out JsonElement snapshotElement) &&
+                snapshotElement.ValueKind == JsonValueKind.Object)
+            {
+                foreach (JsonProperty prop in snapshotElement.EnumerateObject())
+                {
+                    merged[prop.Name] = prop.Value.ValueKind switch
+                    {
+                        JsonValueKind.True => true,
+                        JsonValueKind.False => false,
+                        JsonValueKind.Number => prop.Value.TryGetInt32(out int intVal) ? intVal : (object)prop.Value.GetDouble(),
```
Comment (rishigupta1599) [inline Percy/Percy.cs]: **[Medium] Numeric config values can be silently dropped**

This coerces config numbers `int`-first and falls back to `double`. A config integer above `Int32.MaxValue` (e.g. a large `minHeight`) becomes a `double`, which downstream `TryGetIntFromValue` does not handle, so the value is silently dropped and the fallback is used.

**Suggestion:** Use `TryGetInt64` with an Int32 range check before narrowing, or leave numeric values as the raw `JsonElement` so each consumer parses consistently.

_Reviewer: stack:code-review_

**Review by rishigupta1599 — COMMENTED** [review]
Claude Code Review (automated) — 1 inline finding(s). Full report in the PR comment below. Verdict: Passed.

**Comment by rishigupta1599** [comment]
# Claude Code PR Review

**PR:** https://github.com/percy/percy-playwright-dotnet/pull/220  •  **Head:** 8c2737b  •  **Reviewers:** stack:code-review

## Summary
Adds `MergeSnapshotOptions` to merge `.percy.yml` config `snapshot` keys into per-call snapshot options (per-call wins) before DOM serialization in `Snapshot()`, plus a unit test covering config ⇄ per-snapshot merge precedence. This re-run reviews the current head after the merge-precedence test was added.

## Review Table

| Priority | Category | Check | Status | Notes |
|----------|----------|-------|--------|-------|
| High | Security | No hardcoded secrets or credentials | N/A | Security lens skipped for this change (no secrets touched). |
| High | Security | Authentication/authorization checks present | N/A | Security skipped; no auth surface in diff. |
| High | Security | Input validation and sanitization | N/A | Security skipped; config values are SDK-owned. |
| High | Security | No IDOR — resource ownership validated | N/A | Security skipped; not applicable. |
| High | Security | No SQL injection (parameterized queries) | N/A | Security skipped; no DB/query code. |
| High | Correctness | Logic is correct, handles edge cases | Pass | Merge direction (config < per-call) is correct; verified `IsResponsiveSnapshotCapture` is unchanged and still resolves correctly with the merged dict. |
| High | Correctness | Error handling is explicit, no swallowed exceptions | Pass | No new error paths; existing try/catch in `Snapshot()` intact. |
| High | Correctness | No race conditions or concurrency issues | Pass | `cliConfig` static field sharing predates this PR; no new shared mutable state introduced. |
| Medium | Testing | New code has corresponding tests | Pass | New `MergeSnapshotOptionsTests` covers the core precedence case (config-only key kept, per-call overrides config). |
| Medium | Testing | Error paths and edge cases tested | Fail | Missing coverage: `options == null` (config-only), nested-object config value (`JsonElement` passthrough), numeric coercion, and `cliConfig == null`. |
| Medium | Testing | Existing tests still pass (no regressions) | Pass | Change is additive; existing call sites preserved. |
| Medium | Performance | No N+1 queries or unbounded data fetching | Pass | One bounded dictionary merge per snapshot; negligible. |
| Medium | Performance | Long-running tasks use background jobs | N/A | Not applicable to an SDK helper. |
| Medium | Quality | Follows existing codebase patterns | Pass | `JsonElement`/`ValueKind` switch matches existing config-reading style; test reflection mirrors `CorsFrameTests`. |
| Medium | Quality | Changes are focused (single concern) | Pass | Scoped to config/option merge plus one test. |
| Low | Quality | Meaningful names, no dead code | Fail | `IsResponsiveSnapshotCapture` and `CalculateDefaultHeight` retain `cliConfig` fallbacks now partially redundant since `mergedOptions` already encodes config; latent dead-code/maintainability risk. |
| Low | Quality | Comments explain why, not what | Fail | Intentional asymmetry at `Percy.cs:836` (original `options`, not `mergedOptions`, sent to the CLI `/percy/snapshot`) is undocumented and invites an incorrect "fix". |
| Low | Quality | No unnecessary dependencies added | Pass | No new dependencies. |

## Findings

- **File:** `Percy/Percy.cs:743`
- **Severity:** Medium
- **Reviewer:** stack:code-review
- **Issue:** Numeric config values are coerced `int`-first, falling back to `double` (`TryGetInt32(...) ? intVal : (object)GetDouble()`). A config integer exceeding `Int32.MaxValue` (e.g. an unusually large `minHeight`) becomes a `double`, which downstream `TryGetIntFromValue` (used by `CalculateDefaultHeight`) does not handle — the value is silently dropped and the fallback height is used.
- **Suggestion:** Use `TryGetInt64` with an `Int32` range check before narrowing, or leave numeric values as the raw `JsonElement` (the `_` branch) so each consumer parses consistently.

- **File:** `Percy/Percy.cs:836`
- **Severity:** Low
- **Reviewer:** stack:code-review
- **Issue:** The payload sent to the CLI `/percy/snapshot` endpoint is built from the original unmerged `options`, while `mergedOptions` is used only for `PercyDOM.serialize`. This is the correct/intended model (the CLI applies `.percy.yml` merging server-side; sending `mergedOptions` would double-apply), but the asymmetry is undocumented.
- **Suggestion:** Add a one-line comment at line 836 stating that original options are sent to the CLI deliberately because the CLI does its own config merge.

- **File:** `Percy/Percy.cs:783`
- **Severity:** Low
- **Reviewer:** stack:code-review
- **Issue:** Because `mergedOptions` now already promotes `responsiveSnapshotCapture`/`minHeight`/`readiness` from config, the `cliConfig` fallback branches inside `IsResponsiveSnapshotCapture`, `CalculateDefaultHeight`, and `WaitForReady` are now largely redundant. Behavior is correct today, but the duplicated config-resolution paths are a maintainability hazard.
- **Suggestion:** Optionally simplify these helpers to trust the merged dict as the single source of config truth in a follow-up.

- **File:** `Percy.Test/Percy.Test.cs:1084`
- **Severity:** Medium
- **Reviewer:** stack:code-review
- **Issue:** The single test only covers string+bool precedence. Uncovered: `options == null` (config-only merge), nested-object config value round-tripping as `JsonElement`, numeric coercion, and `cliConfig == null` (merged == options verbatim).
- **Suggestion:** Add the four cases above to lock in the merge contract and guard the numeric-coercion edge case.

---

**Verdict: PASS** — the merge logic is correct and focused with a passing precedence test; remaining items (numeric coercion edge case, added test cases, undocumented CLI-payload asymmetry, redundant config fallbacks) are Medium/Low quality improvements, none High.

**Comment by rishigupta1599** [comment]
# Claude Code PR Review

**PR:** https://github.com/percy/percy-playwright-dotnet/pull/220  •  **Head:** 6c8cb05  •  **Reviewers:** stack:code-review

## Summary
Replaces the shallow `.percy.yml` config ⇄ per-call snapshot-option merge with a recursive **deep merge** (`JsonElementToObjectDeep` + `DeepMerge`) so nested config blocks (e.g. `discovery`) merge leaf-by-leaf with per-call options winning, instead of a per-call object replacing the whole config block. The merged result feeds DOM serialization (`IsResponsiveSnapshotCapture` / `CaptureResponsiveDom` / `GetSerializedDom`). Adds a `MergeSnapshotOptionsTests` class with 2 unit tests (config-only key retention + per-call override; nested deep-merge) — both pass (2/2).

## Review Table

| Priority | Category | Check | Status | Notes |
|----------|----------|-------|--------|-------|
| High | Security | No hardcoded secrets or credentials | N/A | Pure in-memory option-merge logic; no secrets. |
| High | Security | Authentication/authorization checks present | N/A | No auth surface touched. |
| High | Security | Input validation and sanitization | N/A | No external/untrusted input introduced. |
| High | Security | No IDOR — resource ownership validated | N/A | No resource access. |
| High | Security | No SQL injection (parameterized queries) | N/A | No DB / query layer. |
| High | Correctness | Logic is correct, handles edge cases | Pass | `JsonElementToObjectDeep` recurses objects/arrays and maps primitives (int32→int else double, bool, string); `DeepMerge` recurses only when both sides are `Dictionary<string,object>`, else override replaces (arrays/scalars). Per-call wins at leaves. Null options short-circuits to config-only. Verified against JS sdk-utils semantics. |
| High | Correctness | Error handling is explicit, no swallowed exceptions | Pass | Caller `Snapshot()` retains its existing try/catch; merge helpers throw nothing new. Null-config and null-options handled via guards. |
| High | Correctness | No race conditions or concurrency issues | Pass | Reads static `cliConfig` (set at init) and builds fresh dictionaries; no shared mutable state mutated. |
| Medium | Testing | New code has corresponding tests | Pass | Two new unit tests cover the flat-override and nested-deep-merge paths; both pass. |
| Medium | Testing | Error paths and edge cases tested | Pass (minor gap) | Array-replace and double/string primitive conversion not directly asserted; non-blocking test gap. |
| Medium | Testing | Existing tests still pass (no regressions) | Pass | Snapshot wiring change is additive; `Snapshot()` still appends raw `options` to the POST body unchanged. |
| Medium | Performance | No N+1 queries or unbounded data fetching | N/A | No data access; merge is O(size of options). |
| Medium | Performance | Long-running tasks use background jobs | N/A | Synchronous in-memory work only. |
| Medium | Quality | Follows existing codebase patterns | Pass | Mirrors `IsResponsiveSnapshotCapture`'s `cliConfig is JsonElement` access pattern; tests reuse the established reflection technique. |
| Medium | Quality | Changes are focused (single concern) | Pass | Scoped to the config↔per-call merge and its test. |
| Low | Quality | Meaningful names, no dead code | Pass | `JsonElementToObjectDeep` / `DeepMerge` / `MergeSnapshotOptions` are descriptive; no dead code. |
| Low | Quality | Comments explain why, not what | Pass | Comments note the deep-merge intent and the JS sdk-utils parity rationale. |
| Low | Quality | No unnecessary dependencies added | Pass | Uses existing `System.Text.Json`; no new deps. |

## Findings

No blocking findings.

Non-blocking observations (informational, not verdict-affecting):
- **Merged options not included in the POST body** — `Snapshot()` still appends the raw `options` (not `mergedOptions`) to `snapshotOptions` sent to `/percy/snapshot`. Per adjudication this is **INTENDED**: the CLI performs the config merge server-side, so the merged set is only needed locally for DOM serialization. Not a regression.
- **Test-coverage gap** — array-replacement and `double`/`string` primitive conversions aren't directly asserted. Pre-existing-style gap; non-blocking.

---

**Verdict: PASS** — deep-merge logic is correct (per-call wins at leaves, nested objects recurse, arrays/scalars replace), well-tested (2/2 passing), and the only "merged not in POST body" item is intended CLI-side-merge behavior. No new regression.

**Comment by rishigupta1599** [comment]
# Claude Code PR Review

**PR:** https://github.com/percy/percy-playwright-dotnet/pull/220  •  **Head:** b7ba965  •  **Reviewers:** stack:code-review

## Summary
Follow-up fix to the `.percy.yml` config ⇄ per-snapshot deep-merge: `JsonElementToObjectDeep` now converts all-integer JSON arrays (e.g. config `widths`) to `List<int>` instead of `List<object>`, so `ParseWidthsFromOptions`' `IEnumerable<int>` path recognizes config-supplied widths instead of silently dropping them. Adds two xUnit tests for merge precedence and nested deep-merge. Deep-merge semantics otherwise unchanged.

## Review Table

| Priority | Category | Check | Status | Notes |
|----------|----------|-------|--------|-------|
| High | Security | No hardcoded secrets or credentials | N/A | Security lens skipped per task; no secrets in diff. |
| High | Security | Authentication/authorization checks present | N/A | Security lens skipped; no auth surface touched. |
| High | Security | Input validation and sanitization | N/A | Security lens skipped. |
| High | Security | No IDOR — resource ownership validated | N/A | Security lens skipped; no resource access. |
| High | Security | No SQL injection (parameterized queries) | N/A | Security lens skipped; no DB access. |
| High | Correctness | Logic is correct, handles edge cases | Pass | All-int array → `List<int>` via `items.All(i => Number && TryGetInt32)`; mixed/double/empty arrays correctly fall to `List<object>`. DeepMerge: dicts recurse, arrays/scalars replace. `ParseWidthsFromOptions` `IEnumerable<int>` branch now matches (was dropped as `List<object>` before). Verified against source lines 540-558, 157-218. |
| High | Correctness | Error handling is explicit, no swallowed exceptions | Pass | Snapshot path retains its existing try/catch; no new swallowing introduced by the conversion helpers. |
| High | Correctness | No race conditions or concurrency issues | Pass | Pure synchronous transforms over local dictionaries; no shared mutable state added. |
| Medium | Testing | New code has corresponding tests | Pass | Two tests cover config-only-key retention + per-call override and nested deep-merge. No widths-specific test (non-blocking test-gap per adjudication). |
| Medium | Testing | Error paths and edge cases tested | Partial | Happy-path merge covered; empty/mixed array and widths→List<int> conversion not directly asserted. Non-blocking. |
| Medium | Testing | Existing tests still pass (no regressions) | Pass (not run locally) | Could not execute: local dotnet runtime arch mismatch (x86_64 runtime vs arm64). Change is additive + behind merge path; static analysis shows no regression. |
| Medium | Performance | No N+1 queries or unbounded data fetching | N/A | No data fetching; bounded recursion over config tree. |
| Medium | Performance | Long-running tasks use background jobs | N/A | Not applicable to SDK serialization path. |
| Medium | Quality | Follows existing codebase patterns | Pass | Mirrors existing `JsonElement`/reflection conventions (`TryGetIntFromValue`, CorsFrameTests reflection style). |
| Medium | Quality | Changes are focused (single concern) | Pass | Diff is scoped to the List<int> conversion + supporting tests. |
| Low | Quality | Meaningful names, no dead code | Pass | `JsonElementToObjectDeep`, `DeepMerge`, `MergeSnapshotOptions` are clear; no dead code. |
| Low | Quality | Comments explain why, not what | Pass | Comment on the all-int branch explains the `ParseWidthsFromOptions` rationale (the actual bug). |
| Low | Quality | No unnecessary dependencies added | Pass | Uses existing `System.Linq` / `System.Text.Json`; no new deps. |

## Findings

No blocking findings.

Adjudicated as INTENDED / non-blocking (not verdict-failing):
- **Merged options not in POST body** (`Percy/Percy.cs:902-904`): only original per-call `options` are added to `snapshotOptions`; `mergedOptions` (incl. config-only keys like config `widths`) drives DOM/responsive capture but not the POST body. INTENDED — the CLI merges config server-side; config widths correctly drive responsive DOM capture, which is the purpose of the fix.
- **No widths-specific test** + **existing tests not run locally** (dotnet runtime arch mismatch): pre-existing test-gap / environment limitation, non-blocking per adjudication.

Verification performed:
- `List<int>` conversion: confirmed all-int arrays return `List<int>` (Percy.cs:170-176) and that `ParseWidthsFromOptions` (Percy.cs:547-550) now hits the `IEnumerable<int>` branch — previously `List<object>` fell through to `return new List<int>()` (silent drop). Genuine fix, no new regression.
- Deep-merge correctness: `DeepMerge` (Percy.cs:200-218) recurses only when both values are `Dictionary<string,object>`, else override wins; arrays/scalars replace. Matches documented intent.
- Test reflection targets exist with correct signatures: `cliConfig` (private static), `setCliConfig` (internal static, reachable via NonPublic flag), `MergeSnapshotOptions` (private static). Required usings all present.

---

**Verdict: PASS** — follow-up fix is correct; the all-int → List<int> conversion genuinely resolves the silent widths drop, deep-merge semantics are sound, and no new regression introduced.

**Review by Shivanshu-07 — APPROVED** [review]
